src/emulator/remoteconfig/index.ts

```diff
@@ -0,0 +1,301 @@
+    this._emulatorTemplate = newTemplate;
+    this._cloudFunctions.dispatch("update", {
+      description: "Emulator template updated.",
+      updateOrigin: "REMOTE_CONFIG_UPDATE_ORIGIN_UNSPECIFIED",
+      updateTime: new Date().toISOString(),
+      updateType: "REMOTE_CONFIG_UPDATE_TYPE_UNSPECIFIED",
+      updateUser: {
+        email: "emulator@example.com",
+      },
+      versionNumber: 1,
+    });
+  }
+
+  async start(): Promise<void> {
+    const { host, port } = this.getInfo();
+    this._logger.logLabeled("BULLET", "remote config", `Emulator loading...`);
+
+    this._app = await createApp(this.args.projectId, this);
+
+    if (typeof this.args.template === "string") {
+      const templateFile = this.args.template;
+      this.updateTemplateSource(templateFile);
+    } else {
+      this._templateSource = this.args.template;
+    }
+
+    if (!this._templateSource || this._templateSource.files.length === 0) {
+      throw new FirebaseError(
+        "Can not initialize Remote Config emulator without a template source / file."
+      );
+    } else if (this._templateSource.files.length > 1) {
+      throw new FirebaseError(
+        "Can not initialize Remote Config emulator with more than one template source / file."
+      );
+    }
+
+    this.loadTemplate(this._templateSource);
+
+    const templatePath = this._templateSource.files[0].name;
+    this._templateWatcher = chokidar.watch(templatePath, { persistent: true, ignoreInitial: true });
+    this._templateWatcher.on("change", async () => {
+      // There have been some race conditions reported (on Windows) where reading the
+      // file too quickly after the watcher fires results in an empty file being read.
+      // Adding a small delay prevents that at very little cost.
+      await new Promise((res) => setTimeout(res, 5));
+
+      this._logger.logLabeled(
+        "BULLET",
+        "remoteconfig",
+        `Change detected, updating template for Remote Config...`
+      );
+      this.updateTemplateSource(templatePath);
+      this.loadTemplate(this._templateSource);
+    });
+
+    const server = this._app.listen(port, host);
+    this.destroyServer = utils.createDestroyer(server);
+  }
+
+  public loadTemplate(source?: Source): void {
+    if (!source) {
+      source = this._templateSource;
+    }
+    // TODO(kroikie): validate template before using it.
+    this._emulatorTemplate = RemoteConfigEmulator.prepareEmulatorTemplate(
+      JSON.parse(source!.files[0].content)
+    );
+  }
+
+  /**
+   * Update incoming template to be used by the emulator server.
+   * If necessary `!isEmulator` conditional values are added to
+   * each parameter.
+   *
+   * @param template A valid remote config template.
+   */
+  static prepareEmulatorTemplate(template: any): any {
+    const emulatorTemplate = cloneDeep(template);
+    const emulatorParameters = emulatorTemplate["parameters"] || {};
+    for (const parameterName of Object.keys(emulatorParameters)) {
```

Comment:
Can simplify this a lil to `for (const [parameterName, emulatorParameter] of Object.entries(emulatorParameters)) {`